features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/ProtobufMapper.java

```diff
@@ -245,6 +248,9 @@ public ProtobufMapper(EventConfDao eventConfDao, HwEntityDao hwEntityDao, Sessio
         setTimeIfNotNull(event.getTime(), builder::setTime);
         setTimeIfNotNull(event.getCreationTime(), builder::setCreateTime);
 
+        String eventSnmp = event.getSnmp() == null ? null : SnmpInfo.format(event.getSnmp(), EVENT_SNMP_FIELD_SIZE);
```

Comment:
I think we'd be better off to model this with multiple fields or a new object rather than formatting this as a string - how are clients expected to parse this?

Reply:
I lifted this from [here](https://github.com/OpenNMS/opennms/blob/7ecbfee2a8ea39d250ef3982167de39eee34df99/features/events/daemon/src/main/java/org/opennms/netmgt/eventd/processor/HibernateEventWriter.java#L293).   I was trying to match it with `eventSnmp` field from OnmsEvent.

Reply:
Doesn't map well to the protobuf model IMO. If we want to include this information should be break it out as individual fields and save the client from having the parse the string.

Reply:
I agree on having discrete fields like the way `ISnmp` references it within an `IEvent`.

However, the use case for the customer who raised this particular change can retrieve the Trap ID.

OpenNMS doesn't have a field for that, but it can be build by concatenating the [enterprise](https://github.com/OpenNMS/opennms/blob/master/features/events/api/src/main/java/org/opennms/netmgt/events/api/model/ISnmp.java#L40) field (named just `id` for some reason, but called `enterprise` within the Mask element) plus the [specific](https://github.com/OpenNMS/opennms/blob/master/features/events/api/src/main/java/org/opennms/netmgt/events/api/model/ISnmp.java#L42) fields from the `ISnmp` object; makes sense? Although it would be great to combine both, remember that SNMPv2 traps know one OID as the Trap ID, whereas SNMPv1 traps dissect that into "enterprise" + "generic" + "specific".

Also, the customer who requested this last change is using Meridian 2019 (even if the other customer who triggered the JIRA issue is using Horizon). Fortunately, "adding" new fields to a Protobuf definition doesn't alter the contract (the older version of the ".proto" works, and you get the new fields once you update the client code), so I believe it should be fine to rebase that to 2019.

Reply:
Created SnmpInfo in Events.

````
message SnmpInfo {
  string id = 1;
  string id_text = 2;
  string version = 3;
  uint32 specific = 4;
  uint32 generic = 5;
  string community = 6;
}
```